zerver/lib/topic.py

```diff
@@ -270,3 +270,16 @@ def get_topic_history_for_stream(
     cursor.close()
 
     return generate_topic_history_from_db_rows(rows)
+
+
+def get_bare_topic_name(stored_name: str) -> str:
+    """
+    Resolved topics are denoted only by a title change, not by a boolean toggle in a database column. Thus,
+    this method can be used to get the unmangled title of a given topic, whether it is resolved or not. In the
+    event the output of this method differs from the input stored_name, it can be assumed that the topic is
+    resolved.
+    """
+    if stored_name.startswith(RESOLVED_TOPIC_PREFIX):
+        return stored_name.lstrip(RESOLVED_TOPIC_PREFIX)
```

Comment:
How did you select `lstrip` as the string manipulation function to use here? I'm not sure whether its behavior would be correct for (extremely unlikely) topics that start with ✔✔✔, for example.

Reply:
I selected it by crossing brain wires after a while not working in Python! I remembered `lstrip` existed but _not_ its combinatory properties[^1]. Indeed, this should be `.removeprefix` instead, thank you for that catch!

[^1]: https://docs.python.org/3/library/stdtypes.html#str.lstrip

Reply:
Fixed in 5bcff13e20

Reply:
Oh cool, I didn't know about `removeprefix`; I guess another reason to be excited about being able to upgrade Python.